src/main/java/hudson/plugins/ec2/PluginImpl.java

```diff
@@ -77,7 +77,7 @@ public String getDisplayName() {
     @Override
     public void postInitialize() throws IOException {
         // backward compatibility with the legacy class name
-        Jenkins.XSTREAM.alias("hudson.plugins.ec2.EC2Cloud", AmazonEC2Cloud.class);
+        Jenkins.XSTREAM.alias("hudson.plugins.ec2.EC2Cloud", AmazonEC2Cloud.class, EC2Cloud.class);
```

Comment:
Serialize `AmazonEC2Cloud` to `EC2Cloud`, and deserialize `EC2Cloud` to `EC2Cloud` rather than `AmazonEC2Cloud`.

There is a minor issue with this: if creating an `AmazonEC2Cloud` with a Groovy script, it will be serialized to `EC2Cloud` but remain in memory as `AmazonEC2Cloud` until the next Jenkins restart, which will also means JCasC export won't work until the Jenkins restart. This seems acceptable, especially because any user who cares about this behavior can always fix the problem by migrating to `EC2Cloud` in their Groovy script.